### PR TITLE
Ci pipeline

### PR DESCRIPTION
**Pull request by KamilKhairullin**
- npm run lint 
- npm run check (with continue-on-error = true)

P.S: need to merge husky PR and rebase this one to master, then pipeline will be green

### CODE REVIEW

File: .github/workflows/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: pipeline
+
+on:
+  pull_request:
+    branches:
+      - master
+  schedule:
+      - cron:  '30 5,17 * * *'
```
Comment (aabounegm) [inline .github/workflows/pipeline.yml]: Why do we need this as a cron job? It's not like the files' formatting can spontaneously change itself 😅

File: .github/workflows/pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+name: pipeline
+
+on:
+  pull_request:
+    branches:
+      - master
```
Comment (aabounegm) [inline .github/workflows/pipeline.yml]: Why specifically `master`? This could be useful to have on any Pull Request in general

File: .github/workflows/pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+name: pipeline
```
Comment (aabounegm) [inline .github/workflows/pipeline.yml]: Please try to give a more descriptive name

File: .github/workflows/pipeline.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      matrix:
+        os: [ubuntu-latest]
+        node: [14]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@master
+
+      - name: Setup node env
+        uses: actions/setup-node@v2.1.2
```
Comment (aabounegm) [inline .github/workflows/pipeline.yml]: `v3` is already available. `v2.1.2` is really outdated (3 years ago)

File: .github/workflows/pipeline.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        with:
+          node-version: ${{ matrix.node }}
+
+      - name: Cache node_modules
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
```
Comment (aabounegm) [inline .github/workflows/pipeline.yml]: With the latest `setup-node` action version, caching is built-in. You just need to specify the parameter `cache: 'npm'`

File: .github/workflows/pipeline.yml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+      - cron:  '30 5,17 * * *'
+
+jobs:
+  pipeline:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        node: [14]
```
Comment (aabounegm) [inline .github/workflows/pipeline.yml]: The minimal supported Node version is 16 (the recommended is 18)

File: .github/workflows/pipeline.yml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+            ${{ runner.os }}-node-
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Run code style check
+        run: npm run lint  
+        
+      - name: Run Svelte check
+        continue-on-error: true
```
Comment (aabounegm) [inline .github/workflows/pipeline.yml]: What's the use of `continue-on-error` here? We want to fail on error, and it's the last step anyway so it doesn't even make sense.

File: .gitignore [hunk 8]
```diff
@@ -2,6 +2,7 @@
 node_modules
 /build
 /.svelte-kit
+/.github
```
Comment (aabounegm) [inline .gitignore]: Why did you add this folder to gitignore? And it's even already committed

**Review by aabounegm — CHANGES_REQUESTED** [review]

File: .github/workflows/pipeline.yml (hunk 1 above)
Comment (KamilKhairullin) [reply .github/workflows/pipeline.yml]: For fun and cool green pipeline page in presentation =)

File: .github/workflows/pipeline.yml (hunk 7 above)
Comment (KamilKhairullin) [reply .github/workflows/pipeline.yml]: Now in project 20+ Svelte check errors. And we will not able to merge PRs without fixing them

File: .gitignore (hunk 8 above)
Comment (KamilKhairullin) [reply .gitignore]: > Why did you add this folder to gitignore? And it's even already committed

yes, this do not make sense =) Probably, I wanted add this to .prettierignore

File: .github/workflows/pipeline.yml (hunk 1 above)
Comment (aabounegm) [reply .github/workflows/pipeline.yml]: I still think it is unnecessary, and the time itself is arbitrary. We will have the green checkmark without it anyway. You can run it on `push` to `master` as well if you'd like an extra checkmark

File: .github/workflows/quality-control-pr.yml [hunk 9]
```diff
@@ -0,0 +1,32 @@
+name: Quality Controll on pull request
+
+on: 
+  pull_request:
+    types: [opened, synchronize, reopened]
```
Comment (aabounegm) [inline .github/workflows/quality-control-pr.yml]: This is the default value, no need to specify it explicitly

File: .github/workflows/quality-control-pr.yml [hunk 10]
```diff
@@ -0,0 +1,32 @@
+name: Quality Controll on pull request
```
Comment (aabounegm) [inline .github/workflows/quality-control-pr.yml]:
```suggestion
name: Quality Control on pull request
```

File: .prettierignore [hunk 11]
```diff
@@ -3,6 +3,7 @@ node_modules
 /build
 /.svelte-kit
 /package
+/.github
```
Comment (aabounegm) [inline .prettierignore]: Why would you ignore it in Prettier? Prettier can format YAML files, and this folder does not contain any auto-generated files.

**Review by aabounegm — CHANGES_REQUESTED** [review]

File: .github/workflows/quality-control-pr.yml [hunk 12]
```diff
@@ -0,0 +1,31 @@
+name: Quality Control on pull request
+
+on:
+  pull_request:
+    schedule:
+      - cron: '30 5,17 * * *'
```
Comment (aabounegm) [inline .github/workflows/quality-control-pr.yml]: Again, I think this cron schedule is unnecessary, and it even seems to have the wrong indentation. Please remove it

**Review by aabounegm — CHANGES_REQUESTED** [review]

**Review by aabounegm — APPROVED** [review]
